### PR TITLE
Use new searchfields throughout

### PR DESCRIPTION
**Pull request by eddturner**

### CODE REVIEW

File: proteome-rest/src/main/java/org/uniprot/api/proteome/service/GeneCentricSortClause.java [hunk 1]
```diff
@@ -29,23 +33,24 @@ protected Sort createDefaultSort(boolean hasScore) {
...
                                 val.getLeft()
                                         .equals(
-                                                GeneCentricField.Sort.accession_id
-                                                        .getSolrFieldName()))) {
+                                                UniProtSearchFields.GENECENTRIC
+                                                        .getSortFieldFor(ACCESSION_ID)
+                                                        .getName()))) {
             return fieldSortPairs;
         } else {
             List<Pair<String, Sort.Direction>> newFieldSortPairs = new ArrayList<>();
```
Comment (sahmad-ebi) [inline proteome-rest/src/main/java/org/uniprot/api/proteome/service/GeneCentricSortClause.java]: The problem with using Pair is that it is not very obvious what is in the left and what is in the right. No need to make any change. :-)

File: proteome-rest/src/main/java/org/uniprot/api/proteome/service/ProteomeSortClause.java [hunk 2]
```diff
@@ -8,24 +8,30 @@
...
-import org.uniprot.store.search.field.ProteomeField;
+import org.uniprot.store.search.field.UniProtSearchFields;
 
 /**
  * @author jluo
  * @date: 29 Apr 2019
  */
 @Component
 public class ProteomeSortClause extends AbstractSolrSortClause {
+    private static final String UPID = "upid";
```
Comment (sahmad-ebi) [inline proteome-rest/src/main/java/org/uniprot/api/proteome/service/ProteomeSortClause.java]: We should put such string constant in a util class.

File: proteome-rest/src/main/java/org/uniprot/api/proteome/service/GeneCentricSortClause.java [hunk 3]
```diff
@@ -8,17 +8,21 @@
...
-import org.uniprot.store.search.field.GeneCentricField;
+import org.uniprot.store.search.field.UniProtSearchFields;
 
 /**
  * @author jluo
  * @date: 17 May 2019
  */
 @Component
 public class GeneCentricSortClause extends AbstractSolrSortClause {
+    private static final String ACCESSION_ID = "accession_id";
```
[the remark this answers is not in the document]
Comment (eddturner) [reply proteome-rest/src/main/java/org/uniprot/api/proteome/service/GeneCentricSortClause.java]: looking at this `*SortClause` code, to me it looks like all implementations of `AbstractSolrSortClause` are actually doing a similar thing, and probably should be doing the same thing. Maybe we should raise a separate ticket to refactor the sort clause code, and doing so I think this constant would go away.

File: proteome-rest/src/main/java/org/uniprot/api/proteome/service/GeneCentricSortClause.java (hunk 1 above)
Comment (eddturner) [reply proteome-rest/src/main/java/org/uniprot/api/proteome/service/GeneCentricSortClause.java]: Agreed. I would think we should raise a ticket for this and put it on the backlog

**Pull request merged**